diff --git a/hw/i386/xen/xen-hvm.c b/hw/i386/xen/xen-hvm.c
index 9519c33c0..628fe1e32 100644
--- a/hw/i386/xen/xen-hvm.c
+++ b/hw/i386/xen/xen-hvm.c
@@ -1434,6 +1434,10 @@ void xen_hvm_init_pc(PCMachineState *pcms, MemoryRegion **ram_memory)
      */
     qemu_register_wakeup_support();
 
+    if (xen_stubdom_enable()) {
+        xc_set_hvm_param(xen_xc, xen_domid, HVM_PARAM_DM_DOMAIN, DOMID_SELF);
+    }
+
     rc = xen_map_ioreq_server(state);
     if (rc < 0) {
         goto err;
@@ -1510,12 +1514,20 @@ void xen_hvm_init_pc(PCMachineState *pcms, MemoryRegion **ram_memory)
 
     xen_bus_init();
 
+#ifndef CONFIG_STUBDOM
     /* Initialize backend core & drivers */
     if (xen_be_init() != 0) {
         error_report("xen backend core setup failed");
         goto err;
     }
     xen_be_register_common();
+#else
+    xenstore = xs_daemon_open();
+    if (!xenstore) {
+        error_report("can't connect to xenstored");
+        goto err;
+    }
+#endif
 
     QLIST_INIT(&xen_physmap);
     xen_read_physmap(state);
